### PR TITLE
Display and lock

### PR DESCRIPTION
**Pull request by zotakk4o**

### CODE REVIEW

File: Lock.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+		bool locked;
+		int pin;
+	public:
+		Lock(int pin);
+		Lock(const Lock&) = delete;
+		Lock& operator=(const Lock&) = delete;
+
+		void init();
+		void unlock(int timeBeforeLock = 0);
+		void lock();
```
Comment (yalishanda42) [inline Lock.hpp:15]: I think in this PR we should also call these classes and methods inside the `.ino` file

File: DisplayConfig.hpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#include "Display.hpp"
+
+class DisplayConfig {
+	public:
+		static void displayTemperatureAndHumidity(Display& display, int temperature, int humidity);
+		static void displayAdminKeyMessage(Display& display);
+		static void displayNewKeyAuthMessage(Display& display);
+		static void displayKeyAddedMessage(Display& display);
+		static void displayDoorLockedMessage(Display& display);
+		static void displayDoorUnlockedMessage(Display& display);
```
Comment (yalishanda42) [inline DisplayConfig.hpp]: What if we instead have this repeating `Display&` parameter as a member of the class instead and have all these methods be non-static?
Also please remove `StateManager::getStateMessage` as it won't be needed and use this `DisplayConfig` in the `.ino` file to setup the display accordingly

**Review by yalishanda42 — CHANGES_REQUESTED** [review]

File: DisplayConfig.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#ifndef DISPLAYCONFIG_HPP
+#define DISPLAYCONFIG_HPP
```
Comment (ihristova11) [inline DisplayConfig.hpp]: Maybe `__DISPLAY_CONFIG__HPP__` will be better, just for consistency :)

File: Lock.hpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#ifndef LOCK_HPP
+#define LOCK_HPP
```
Comment (ihristova11) [inline Lock.hpp]: `__LOCK__HPP__`, just for consistency :)

File: Display.hpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#ifndef DISPLAY_HPP
+#define DISPLAY_HPP
```
Comment (ihristova11) [inline Display.hpp]: `__DISPLAY_HPP__`, just for consistency :)

**Pull request merged**